pkg/driver/driver_alicloud.go

```diff
@@ -0,0 +1,287 @@
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/utils"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+)
+
+// AlicloudDriver is the driver struct for holding Alicloud machine information
+type AlicloudDriver struct {
+	AlicloudMachineClass *v1alpha1.AlicloudMachineClass
+	CloudConfig        *corev1.Secret
+	UserData           string
+	MachineID          string
+	MachineName        string
+}
+
+// runInstanceRequest is the request struct for api RunInstances
+type runInstancesRequest struct {
+	*requests.RpcRequest
+	InstanceName            string           `position:"Query" name:"InstanceName"`
+	Description             string           `position:"Query" name:"Description"`
+	UserData                string           `position:"Query" name:"UserData"`
+	ClusterId               string           `position:"Query" name:"ClusterId"`
+	ClientToken             string           `position:"Query" name:"ClientToken"`
+	ImageId                 string           `position:"Query" name:"ImageId"`
+	InstanceType            string           `position:"Query" name:"InstanceType"`
+	ZoneId                  string           `position:"Query" name:"ZoneId"`
+	SecurityGroupId         string           `position:"Query" name:"SecurityGroupId"`
+	VSwitchId               string           `position:"Query" name:"VSwitchId"`
+	PrivateIpAddress        string           `position:"Query" name:"PrivateIpAddress"`
+	InnerIpAddress          string           `position:"Query" name:"InnerIpAddress"`
+	SystemDiskSize          requests.Integer `position:"Query" name:"SystemDisk.Size"`
+	SystemDiskCategory      string           `position:"Query" name:"SystemDisk.Category"`
+	InstanceChargeType      string           `position:"Query" name:"InstanceChargeType"`
+	InternetChargeType      string           `position:"Query" name:"InternetChargeType"`
+	InternetMaxBandwidthIn  requests.Integer `position:"Query" name:"InternetMaxBandwidthIn"`
+	InternetMaxBandwidthOut requests.Integer `position:"Query" name:"InternetMaxBandwidthOut"`
+	SpotStrategy            string           `position:"Query" name:"SpotStrategy"`
+	IoOptimized             string           `position:"Query" name:"IoOptimized"`
+	Tag1Key                 string           `position:"Query" name:"Tag.1.Key"`
+	Tag2Key                 string           `position:"Query" name:"Tag.2.Key"`
+	Tag3Key                 string           `position:"Query" name:"Tag.3.Key"`
+	Tag4Key                 string           `position:"Query" name:"Tag.4.Key"`
+	Tag5Key                 string           `position:"Query" name:"Tag.5.Key"`
+	Tag1Value               string           `position:"Query" name:"Tag.1.Value"`
+	Tag2Value               string           `position:"Query" name:"Tag.2.Value"`
+	Tag3Value               string           `position:"Query" name:"Tag.3.Value"`
+	Tag4Value               string           `position:"Query" name:"Tag.4.Value"`
+	Tag5Value               string           `position:"Query" name:"Tag.5.Value"`
```

Comment:
Can we please have an array of tags - as mentioned in the previous comment ? Looks like this will prevent us from having more than 5 tags.

Reply:
noted, will change.